plan/cache/key.go

```diff
@@ -93,3 +89,38 @@ func NewSQLCacheKey(sessionVars *variable.SessionVars, sql string, schemaVersion
 		readOnly:       readOnly,
 	}
 }
+
+type pstmtPlanCacheKey struct {
```

Comment:
should we also consider timezone, sql mode, snapshot for prepared plan cache ?

Reply:
It is possible, but I think that it is not necessary because the plans are valid only in a local session.